submarine-server/server-submitter/submitter-k8s/src/main/java/org/apache/submarine/server/submitter/k8s/K8sSubmitter.java

```diff
@@ -522,16 +522,39 @@ public Notebook deleteNotebook(NotebookSpec spec) throws SubmarineRuntimeExcepti
   public void createServe(ServeSpec spec)
       throws SubmarineRuntimeException {
     SeldonDeployment seldonDeployment = parseServeSpec(spec);
-
+    IstioVirtualService istioVirtualService = new IstioVirtualService(spec.getModelName(),
+        spec.getModelVersion());
     try {
       api.createNamespacedCustomObject(seldonDeployment.getGroup(),
-              seldonDeployment.getVersion(),
+               seldonDeployment.getVersion(),
+               "default",
+               seldonDeployment.getPlural(),
+               seldonDeployment,
+               "true");
+    } catch (ApiException e) {
+      LOG.error(e.getMessage(), e);
+      throw new SubmarineRuntimeException(e.getCode(), e.getMessage());
+    }
+    try {
+      api.createNamespacedCustomObject(istioVirtualService.getGroup(),
+              istioVirtualService.getVersion(),
               "default",
-              seldonDeployment.getPlural(),
-              seldonDeployment,
+              istioVirtualService.getPlural(),
+              istioVirtualService,
               "true");
     } catch (ApiException e) {
       LOG.error(e.getMessage(), e);
+      try {
+        api.deleteNamespacedCustomObject(seldonDeployment.getGroup(),
+              seldonDeployment.getVersion(),
+              "default",
+              seldonDeployment.getPlural(),
+              seldonDeployment.getMetadata().getName(),
+              new V1DeleteOptionsBuilder().withApiVersion(seldonDeployment.getApiVersion()).build(),
+              null, null, null);
+      } catch (ApiException e1) {
+        LOG.error(e1.getMessage(), e1);
+      }
```

Comment:
Why do we need to delete the object when an error happens? Does it sometimes create the resource when an error happens?